libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -362,8 +362,20 @@ public void run() {
                 } catch (Exception e) {
                     SmartSyncLogger.e(TAG, "Exception thrown in runSync", e);
 
-                    //Set error message to sync state
-                    sync.setError(e.getMessage());
+                    //Get error from API respond
+                    String errorMessage = e.getMessage();
+                    if(errorMessage != null){
+                        try {
+                            //Parse respond string to JSONArray
+                            JSONArray errorInJSON = new JSONArray(errorMessage);
+
+                            //Set error message from first object in JSONArray to sync state
+                            sync.setError(errorInJSON.getJSONObject(0));
+                        } catch (JSONException jsonException) {
+                            jsonException.printStackTrace();
```

Comment:
Replace this line with:
```
SmartSyncLogger.e(TAG, "Exception thrown while parsing JSON", jsonException);
```

Reply:
Resolved in 3d45638d9478cc7560c9ebcce4677a7c761fc26b